File: src/CareerCompass.Services/Models/User/MasterData.cs
using CareerCompass.DataContext.Enums;
using CareerCompass.Services.Models.Enums;

namespace CareerCompass.Services.Models.User
{
    public class MasterData
    {
        public string FirstName { get; set; } = string.Empty;

        public string LastName { get; set; } = string.Empty;

        public string? FieldOfStudy { get; set; }

        public List<EnumModel> Skills { get; set; } = new List<EnumModel>();

        public List<EnumModel> AreasOfInterest { get; set; } = new List<EnumModel>();

        public EnumModel CareerGoal { get; set; }

        public List<SchoolYear> SchoolYears { get; set; } = new List<SchoolYear>();
    }
}

File: src/CareerCompass.UI/src/models/master-data.ts
import { EnumModel, SchoolYear } from "./user";

export interface MasterData {
    firstName: string;
    lastName: string;
    fieldOfStudy: string | null;
    skills: EnumModel[];
    areasOfInterest: EnumModel[];
    careerGoal: EnumModel | null;
    schoolYears: SchoolYear[];
}